addons/account/models/account_bank_statement.py

```diff
@@ -178,10 +178,38 @@ def _compute_journal_id(self):
 
     @api.depends('balance_end_real', 'balance_end')
     def _compute_is_complete(self):
+        incomplete_statements = self.filtered(lambda stmt: not stmt.is_complete)
+
         for stmt in self:
             stmt.is_complete = stmt.line_ids and stmt.currency_id.compare_amounts(
                 stmt.balance_end, stmt.balance_end_real) == 0
 
+        # Process the bank statement report.
+        complete_stmts = incomplete_statements.filtered('is_complete')
+        if self.ids and complete_stmts:
+            IrActionsReport = self.env['ir.actions.report'].sudo()
+            statement_report = self.env.ref('account.action_report_account_statement').sudo()
+
+            for statement in complete_stmts:
+                attachment_name = _("Bank Statement %s.pdf", statement.name) if statement.name else _("Bank Statement.pdf")
+
+                # Find the existing attachment.
+                existing_attachment = statement.attachment_ids.filtered(lambda x: x.name == attachment_name)
+
+                # Create / update the document.
+                content, _content_type = IrActionsReport._render_qweb_pdf(statement_report, res_ids=statement.ids)
+                if existing_attachment:
+                    existing_attachment.raw = content
+                else:
+                    statement.attachment_ids |= self.env['ir.attachment'].create({
```

Comment:
We're doing this in a compute ?! O.O Is there really no other way ?

Why do we want this anyway ? Just to have the side panel with the content of the statement when clicking on it in list view ? Why not relying on a classical form view, then ?

Reply:
Because you don't have a "manual trigger" from the user anymore. Before that, it was when clicking on "post". This is no longer the case since the state no longer exists. I didn't find any other way outside this compute